src/main/java/com/amazon/opendistroforelasticsearch/ad/task/ADBatchTaskRunner.java

```diff
@@ -188,47 +394,181 @@ public ADBatchTaskRunner(
-                        .info(
-                            "execute AD task {} locally on node {} for detector {}",
-                            adTask.getTaskId(),
-                            node.getId(),
-                            adTask.getDetectorId()
-                        );
-                    startADBatchTask(adTask, false, transportService, delegatedListener);
+            if (adTask.isEntityTask()) {
+                // When reach this line, it means entity task failed to start on worker node
+                if (adTaskManager.isRetryableError(adTask.getError())
+                    && !adTaskCacheManager.exceedRetryLimit(adTask.getDetectorId(), adTask.getTaskId())) {
+                    // If the error is retryable, move entity from temp queue to the end of pending queue
+                    adTaskCacheManager.pushBackEntity(adTask.getTaskId(), adTask.getDetectorId(), adTask.getEntity().get(0).getValue());
                 } else {
-                    // Execute batch task remotely
-                    logger
-                        .info(
-                            "execute AD task {} remotely on node {} for detector {}",
-                            adTask.getTaskId(),
-                            node.getId(),
-                            adTask.getDetectorId()
-                        );
-                    transportService
-                        .sendRequest(
-                            node,
-                            ADBatchTaskRemoteExecutionAction.NAME,
-                            new ADBatchAnomalyResultRequest(adTask),
-                            option,
-                            new ActionListenerResponseHandler<>(delegatedListener, ADBatchAnomalyResultResponse::new)
-                        );
+                    adTaskCacheManager.removeEntity(adTask.getDetectorId(), adTask.getEntity().get(0).getValue());
+                    logger.warn("Entity task failed, task id: {}", adTask.getTaskId());
                 }
-            }, e -> delegatedListener.onFailure(e))), e -> delegatedListener.onFailure(e)));
+                // Sleep some time before polling next entity task.
+                waitBeforeNextEntity(SLEEP_TIME_FOR_NEXT_ENTITY_TASK_IN_MILIS);
+                adTaskManager.entityTaskDone(adTask, e, transportService);
+                startNewEntityTaskLane(adTask, transportService);
+            }
+        });
+
+        ThreadedActionListener threadedActionListener = new ThreadedActionListener<>(
+            logger,
+            threadPool,
+            AD_BATCH_TASK_THREAD_POOL_NAME,
+            actionListener,
+            false
+        );
+        return threadedActionListener;
+    }
+
+    private void waitBeforeNextEntity(long time) {
+        try {
+            Thread.sleep(time);
+        } catch (InterruptedException interruptedException) {
+            logger.warn("Exception while waiting", interruptedException);
+        }
+    }
+
+    private void executeSingleEntityTask(
```

Comment:
The method not only execute, but also forward.  Can we rename the method to forwardOrExecuteEntityTask?

Reply:
Will fix in new PR